### PR TITLE
React dependencies need updating

### LINKED ISSUE #13525: React dependencies need updating
##### **Overview of the issue**

Create a React project with the `main` branch and you'll see there's quite a few vulnerabilities.

```
found 38 vulnerabilities (32 low, 6 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

##### **Motivation for or Use Case**

I like to show how JHipster is up-to-date and doesn't have vulnerabilities.

##### **Reproduce the error**

```
wget https://raw.githubusercontent.com/jhipster/jdl-samples/main/blog-oauth2.jdl
# edit blog-oauth2.jdl and change clientFramework to react
jhipster jdl blog-oauth2.jdl
```

You'll see the following after `npm install`:

```
found 38 vulnerabilities (32 low, 6 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

##### **Suggest a Fix**

Running `ncu -u` shows the following updates available:

```
 @fortawesome/fontawesome-svg-core          1.2.30  →    1.2.34
 @fortawesome/free-solid-svg-icons          5.14.0  →    5.15.2
 @fortawesome/react-fontawesome             0.1.12  →    0.1.14
 axios                                      0.19.2  →    0.21.1
 bootstrap                                   4.4.1  →     4.5.3
 lodash                                    4.17.15  →   4.17.20
 react-redux                                 7.2.0  →     7.2.2
 react-redux-loading-bar                     4.6.0  →     5.0.0
 react-router-dom                            5.1.2  →     5.2.0
 react-toastify                              5.5.0  →     6.2.0
 react-transition-group                      4.3.0  →     4.4.1
 reactstrap                                  8.6.0  →     8.8.1
 redux-devtools                              3.5.0  →     3.7.0
 redux-devtools-dock-monitor                 1.1.3  →     1.2.0
 redux-devtools-log-monitor                  1.4.0  →     2.1.0
 tslib                                      1.11.1  →     2.1.0
 uuid                                        7.0.3  →     8.3.2
 @openapitools/openapi-generator-cli  1.0.13-4.3.1  →    2.1.16
 @testing-library/react                     10.4.8  →    11.2.3
 @types/chai                                4.2.11  →    4.2.14
 @types/jest                               26.0.10  →   26.0.20
 @types/lodash                            4.14.150  →  4.14.167
 @types/mocha                                7.0.2  →     8.2.0
 @types/node                               13.13.4  →  14.14.21
 @types/react                              16.9.34  →    17.0.0
 @types/react-dom                           16.9.6  →    17.0.0
 @types/react-redux                          7.1.7  →    7.1.15
 @types/react-router-dom                     5.1.5  →     5.1.7
 @types/selenium-webdriver                   4.0.9  →    4.0.11
 @types/webpack-env                         1.15.2  →    1.16.0
 @typescript-eslint/eslint-plugin            4.8.0  →    4.13.0
 @typescript-eslint/parser                   4.8.0  →    4.13.0
 autoprefixer                                9.7.6  →    10.2.1
 browser-sync                               2.26.7  →   2.26.13
 browser-sync-webpack-plugin                 2.2.2  →     2.3.0
 copy-webpack-plugin                         6.0.3  →     7.0.0
 core-js                                     3.6.5  →     3.8.2
 cross-env                                   7.0.2  →     7.0.3
 css-loader                                  3.5.3  →     5.0.1
 eslint                                     7.13.0  →    7.17.0
 eslint-config-prettier                     6.15.0  →     7.1.0
 eslint-plugin-react                        7.21.5  →    7.22.0
 eslint-webpack-plugin                       2.3.0  →     2.4.1
 file-loader                                 6.0.0  →     6.2.0
 fork-ts-checker-webpack-plugin              4.1.3  →     6.1.0
 html-webpack-plugin                         4.3.0  →     4.5.1
 jest                                       26.4.2  →    26.6.3
 jest-junit                                 11.1.0  →    12.0.0
 mini-css-extract-plugin                     0.9.0  →     1.3.4
 mocha                                       7.1.2  →     8.2.1
 optimize-css-assets-webpack-plugin          5.0.3  →     5.0.4
 postcss-loader                              3.0.0  →     4.1.0
 protractor                                  5.4.4  →     7.0.0
 sass                                       1.26.5  →    1.32.4
 sass-loader                                 8.0.2  →    10.1.1
 sinon                                       9.0.2  →     9.2.3
 source-map-loader                           0.2.4  →     2.0.0
 style-loader                                1.2.0  →     2.0.0
 swagger-ui-dist                            3.25.1  →    3.40.0
 terser-webpack-plugin                       2.3.6  →     5.1.1
 thread-loader                               2.1.3  →     3.0.1
 ts-jest                                    26.4.3  →    26.4.4
 ts-loader                                   7.0.1  →    8.0.14
 ts-node                                     8.9.1  →     9.1.1
 webdriver-manager                          12.1.7  →    12.1.8
 webpack                                    4.46.0  →    5.14.0
 webpack-cli                                3.3.11  →     4.3.1
 webpack-dev-server                         3.10.3  →    3.11.2
 webpack-merge                               5.0.9  →     5.7.3
 webpack-notifier                            1.8.0  →    1.12.0
 workbox-webpack-plugin                      5.1.3  →     6.0.2
```

##### **JHipster Version(s)**

```
blog@0.0.1-SNAPSHOT /Users/mraible/dev/react-protractor
└── generator-jhipster@7.0.0-beta.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "authenticationType": "oauth2",
    "cacheProvider": "hazelcast",
    "clientFramework": "react",
    "serverPort": "8080",
    "serviceDiscoveryType": false,
    "skipUserManagement": true,
    "withAdminUi": true,
    "baseName": "blog",
    "buildTool": "maven",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": ["en"],
    "messageBroker": false,
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "skipClient": false,
    "testFrameworks": ["protractor"],
    "websocket": false,
    "applicationType": "monolith",
    "packageName": "com.jhipster.demo.blog",
    "packageFolder": "com/jhipster/demo/blog",
    "jhipsterVersion": "7.0.0-beta.0",
    "skipServer": false,
    "clientPackageManager": "npm",
    "dtoSuffix": "DTO",
    "entitySuffix": "",
    "reactive": false,
    "clientTheme": "none",
    "clientThemeVariant": "",
    "entities": ["Blog", "Post", "Tag"],
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "nativeLanguage": "en",
    "creationTimestamp": 1610661173008,
    "lastLiquibaseTimestamp": 1610661353000
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}
entity Post {
  title String required
  content TextBlob required
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}
relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

paginate Post, Tag with infinite-scroll

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

git version 2.24.3 (Apple Git-128)

node: v14.15.0

npm: 6.14.11

yeoman: 3.1.1

Docker version 20.10.2, build 2291f61

docker-compose version 1.27.4, build 40524192

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #13525 

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/package.json.ejs [hunk 1]
```diff
@@ -26,138 +26,138 @@
...
-    "tslib": "1.11.1",
-    "uuid": "7.0.3"<% if (websocket === 'spring-websocket') { %>,
-    "webstomp-client": "1.2.6"
+    "tslib": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "uuid": "VERSION_MANAGED_BY_CLIENT_REACT"<% if (websocket === 'spring-websocket') { %>,
+    "webstomp-client": "VERSION_MANAGED_BY_CLIENT_REACT"
     <%_ } _%>
   },
   "devDependencies": {
-    "@openapitools/openapi-generator-cli": "1.0.13-4.3.1",
```
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:71]: Keep openapi fixed.
See https://github.com/jhipster/generator-jhipster/pull/13528

File: generators/client/templates/react/package.json.ejs [hunk 2]
```diff
@@ -26,138 +26,138 @@
...
+    "redux-devtools": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "redux-devtools-dock-monitor": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "redux-devtools-log-monitor": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "redux-promise-middleware": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "redux-thunk": "VERSION_MANAGED_BY_CLIENT_REACT",
     <%_ if (websocket === 'spring-websocket') { _%>
-    "rxjs": "6.4.0",
-    "sockjs-client": "1.3.0",
+    "rxjs": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "sockjs-client": "VERSION_MANAGED_BY_CLIENT_REACT",
```
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:60]: Better to keep dependencies without tests fixed.
There is no test for websocket.

File: generators/client/templates/react/package.json.ejs [hunk 3]
```diff
@@ -26,138 +26,138 @@
...
     <%_ if (skipServer) { _%>
-    "sonar-scanner": "3.1.0",
+    "sonar-scanner": "VERSION_MANAGED_BY_CLIENT_REACT",
     <%_ } _%>
-    "tslib": "1.11.1",
-    "uuid": "7.0.3"<% if (websocket === 'spring-websocket') { %>,
-    "webstomp-client": "1.2.6"
+    "tslib": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "uuid": "VERSION_MANAGED_BY_CLIENT_REACT"<% if (websocket === 'spring-websocket') { %>,
+    "webstomp-client": "VERSION_MANAGED_BY_CLIENT_REACT"
```
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:67]: Same as above

File: generators/client/templates/react/package.json.ejs [hunk 4]
```diff
@@ -26,138 +26,138 @@
...
-    "mini-css-extract-plugin": "0.9.0",
+    "mini-css-extract-plugin": "VERSION_MANAGED_BY_CLIENT_REACT",
     <%_ if (protractorTests) { _%>
     "mocha": "7.1.2",
     <%_ } _%>
-    "optimize-css-assets-webpack-plugin": "5.0.3",
-    "postcss-loader": "3.0.0",
+    "optimize-css-assets-webpack-plugin": "VERSION_MANAGED_BY_CLIENT_REACT",
+    "postcss-loader": "VERSION_MANAGED_BY_CLIENT_REACT",
     "prettier": "<%= PRETTIER_VERSION %>",
```
Comment (mshima) [inline generators/client/templates/react/package.json.ejs:137]: Add prettier and prettier Java to dependabot

File: generators/client/templates/react/package.json.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs:71]: ok

File: generators/client/templates/react/package.json.ejs (hunk 2 above)
Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs:60]: ok

File: generators/client/templates/react/package.json.ejs (hunk 3 above)
Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs:67]: ok

File: generators/client/templates/react/package.json.ejs (hunk 4 above)
Comment (qmonmert) [reply generators/client/templates/react/package.json.ejs:137]: ok

**Pull request merged**